### ISSUE TITLE
Feature: bind mount dir in a subdir of an already bind mounted dir

### ISSUE THREAD

**Issue by ghuls** [apptainer/singularity#2410]
It would be nice to be able to bind mount a directory in a subdirectory of an already bind mounted directory.

I would like to have a singularity container with a clean (persistent0 home dir (so settings don't conflict with with application version in the container), but with some directories from the real home directory (like ssh settings) bind mounted in the clean home directory.

```bash
# Create fake home dir.
mkdir /fake_home_dir

#   - Bind fake home dir (to which settings will be saved) to $HOME in the container.
#   - Bind real .ssh directory in this fake home directory.
#   - Bind real home directory as HOME inside the fake home dir in the container.
singularity exec \
    --contain \
    --home /fake_home_dir:${HOME} \
    --bind ${HOME}/.ssh:${HOME}/.ssh:ro \
    --bind ${HOME}:${HOME}/HOME
    container.sif
```

At the moment singularity does not allow it, as it complains (debug mode) that ${HOME} is already mounted, so it does not mount *.ssh* and *HOME* inside the container.

**Comment by cclerget** [comment]
@ghuls Yes the code checking already mounted points is too restrictive and need some rework to allow what you want to do, unfortunately there is a thing that won't change, the fact to create directories outside of container image like in your example the `${HOME}/HOME` directory, for security reasons Singularity can't create them.

**Comment by ghuls** [comment]
It is no problem for me to create them outside of singularity.

I now found a workaround, but it is a bit ugly.
Mount those extra dirs outside of an existing one and create a symlink.

```bash
singularity exec \
    --contain \
    --home /fake_home_dir:${HOME} \
    --bind ${HOME}/.ssh:/SSH:ro \
    --bind ${HOME}:/HOME
    container.sif

ln -s /SSH ${HOME}/.ssh
ln -s /HOME ${HOME}/HOME
```

**Comment by EvanTheB** [comment]
Fair enough that this usage is not yet supported, (I hope it will be), but can this be at least made an error?

If I am requesting a bind mount and singularity cannot provide it, that should be an error! Or at least detectable.

**Comment by EvanTheB** [comment]
https://github.com/sylabs/singularity/issues/2085
https://github.com/sylabs/singularity/issues/1833
```

evaben@login-fpga ~> singularity shell -C --bind $HOME/.ssh/id_rsa.pub docker://debian 
Singularity debian_latest.sif:~> pwd
/home/evaben
Singularity debian_latest.sif:~> ls -a
.  ..
Singularity debian_latest.sif:~> ls -a .ssh/id_rsa.pub
ls: cannot access '.ssh/id_rsa.pub': No such file or directory
Singularity debian_latest.sif:~> exit

```

**Comment by cclerget** [comment]
Issue fixed by #3970
